Framework/Algorithms/test/CorelliPowderCalibrationDatabaseTest.h

```diff
@@ -0,0 +1,431 @@
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+
+    EventWorkspace_sptr ws;
+    ws = AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+        "CorelliPowderCalibrationDatabaseTest_OutputWS");
+
+    // Add property start_time
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    return ws;
+  }
+
+  /**
+   * @brief Create Test Calibration TableWorkspace
+   * @param outWSName
+   * @return
+   */
+  TableWorkspace_sptr
+  createTestCalibrationTableWorkspace(const std::string &outWSName) {
+
+    ITableWorkspace_sptr itablews = WorkspaceFactory::Instance().createTable();
+    AnalysisDataService::Instance().addOrReplace(outWSName, itablews);
+
+    TableWorkspace_sptr tablews =
+        std::dynamic_pointer_cast<TableWorkspace>(itablews);
+    TS_ASSERT(tablews);
+
+    // Set up columns
+    for (size_t i = 0;
+         i < CorelliCalibration::calibrationTableColumnNames.size(); ++i) {
+      std::string colname = CorelliCalibration::calibrationTableColumnNames[i];
+      std::string type = CorelliCalibration::calibrationTableColumnTypes[i];
+      tablews->addColumn(type, colname);
+    }
+
+    std::cout << "[DEBUG 0] Table workspace rows: " << tablews->rowCount()
+              << "\n";
+
+    // append rows
+    Mantid::API::TableRow sourceRow = tablews->appendRow();
+    sourceRow << "source" << 0. << 0. << -15.560 << 0. << 0. << 0. << 0.;
+    Mantid::API::TableRow sampleRow = tablews->appendRow();
+    sampleRow << "sample" << 0.0001 << -0.0002 << 0.003 << 0. << 0. << 0. << 0.;
+    Mantid::API::TableRow bank1Row = tablews->appendRow();
+    bank1Row << "bank1" << 0.9678 << 0.0056 << 0.0003 << 0.4563 << -0.9999
+             << 0.3424 << 5.67;
+
+    std::cout << "[DEBUG 1] Table workspace rows: " << tablews->rowCount()
+              << "\n";
+
+    return tablews;
+  }
+
+  /**
+   * @brief Create an incompatile table workspace for algorithm to throw
+   * exception
+   * @param outWSName
+   * @return
+   */
+  TableWorkspace_sptr
+  createIncorrectTestCalibrationTableWorkspace(const std::string &outWSName) {
+    // Create table workspace
+    ITableWorkspace_sptr itablews = WorkspaceFactory::Instance().createTable();
+    AnalysisDataService::Instance().addOrReplace(outWSName, itablews);
+
+    TableWorkspace_sptr tablews =
+        std::dynamic_pointer_cast<TableWorkspace>(itablews);
+    TS_ASSERT(tablews);
+
+    // Set up columns
+    for (size_t i = 0;
+         i < CorelliCalibration::calibrationTableColumnNames.size() - 1; ++i) {
+      std::string colname = CorelliCalibration::calibrationTableColumnNames[i];
+      std::string type = CorelliCalibration::calibrationTableColumnTypes[i];
+      tablews->addColumn(type, colname);
+    }
+
+    // append rows
+    Mantid::API::TableRow sourceRow = tablews->appendRow();
+    sourceRow << "source" << 0. << 0. << -15.560 << 0. << 0. << 0.;
```

Comment:
to be consistent, can you use CORELLI's component names for source and sample ("moderator" and "sample-position")?